### ISSUE TITLE
Config generated by debconf doesn't resemble --generate-config output

### ISSUE THREAD

**Issue by ara4n** [matrix-org/package-synapse-debian#22]
It feels like a recipe for disaster and confusion that the output of debconf doesn't remotely resemble what you get from --generate-config.  Surely we should sync them (especially if generate-config overwrites the config, and so undoes debconf's work).

**Comment by ara4n** [comment]
It also doesn't sound like it generates TLS or signing keys, causing the user to run --generate-config, and thence pain.

**Comment by richvdh** [comment]
The service start script generates the TLS and signing keys, so there's no need for users to try and generate them manually.

--generate-config won't overwrite any existing files.

I think the solution is to stop encouraging users to run it every time they hit any sort of problem, rather than trying to make it completely compatible with the debian-installed default. https://github.com/matrix-org/synapse/pull/2551 includes an attempt at this.

In short: can close this please?

**Comment by ara4n** [comment]
so this stops us from encouraging users to run --generate-config whenever something goes wrong.

but we still have the problem that debconf produces a very stale config which doesn't resemble the up-to-date output you'd get from a --generate-config?

**Comment by richvdh** [comment]
Transferring this to synapse repo, since the packaging is maintained there (and it is still a problem in matrix-synapse-py3)
